ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -301,40 +308,50 @@ void validate() throws IngestionClientException {
 
         String mappingReference = ingestionMapping.getIngestionMappingReference();
```

Comment:
Not exactly related to this PR specfically, but have you seen my PR in node -https://github.com/Azure/azure-kusto-node/pull/154/files?
The column mappings didn't work correctly, so I took the effort to also make them safer by introducing a type for each type of mapping.
Do you think something like that will work here?

Reply:
I hadn't seen that PR. I assume you're referring to `ColumnMapping.isValid()` - what's the potential pitfall there?

Reply:
It's about rather than having one `ColumnMapping` type, having a type for each kind of mapping (E.g `JsonColumnMapping`) that each of them supports only the feature of that column mapping (stuff like transforms and paths), and have their kind initialized correctly.